hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java

```diff
@@ -268,7 +279,8 @@ public Vertex onCalc(CalcPhysicalRel rel) {
             Expression<Boolean> filterExpr = dagBuildContext.convertFilter(rel.filter());
             vertex = dag.newUniqueVertex("Calc", mapUsingServiceP(
                     ServiceFactories.nonSharedService(ctx ->
-                            ExpressionUtil.calcFn(projection, filterExpr, ExpressionEvalContext.from(ctx))),
+                                    ExpressionUtil.calcFn(projection, filterExpr, ExpressionEvalContext.from(ctx)))
+                            .setCooperative(projection.stream().allMatch(Expression::isCooperative)),
```

Comment:
We must have this everywhere we evaluate expressions in a processor. For example 5 lines below and multitude of other places.